fvm/src/kernel/default.rs

```diff
@@ -1114,18 +1107,14 @@ fn get_required_padding(
 
 fn to_fil_public_replica_infos(
     src: &[SectorInfo],
-    typ: ProofType,
+    typ: RegisteredPoStProof,
 ) -> Result<BTreeMap<SectorId, PublicReplicaInfo>> {
     let replicas = src
         .iter()
         .map::<core::result::Result<(SectorId, PublicReplicaInfo), String>, _>(
             |sector_info: &SectorInfo| {
                 let commr = commcid::cid_to_replica_commitment_v1(&sector_info.sealed_cid)?;
-                let proof = match typ {
-                    ProofType::Winning => sector_info.proof.registered_winning_post_proof()?,
-                    ProofType::Window => sector_info.proof.registered_window_post_proof()?,
-                };
```

Comment:
Is there some way to quickly check that the proof type is valid for the given sectors?